app/controllers/data_sources_controller.rb

```diff
@@ -18,6 +18,7 @@ def clear_and_reload
   
   def delete
     source = DataSource.find_by id: params[:id]
+    create_action source,'DELETE'
```

Comment:
Might want to make this conditional on the success of `source.delete` (as you did for the update and create actions)